libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -337,7 +335,44 @@ private void syncUp(SyncState sync, SyncUpdateCallback callback) throws Exceptio
         }
 	}
 
-    private boolean syncUpOneRecord(String soupName, List<String> fieldlist, JSONObject record) throws JSONException, IOException {
+    private boolean isNewerThanServer(String objectType, String objectId,
```

Comment:
We could do group of records with a selectLastModifiedDate where id in (...) but there is a limit to the length of a soql query, so one might have to do several round trips if many records were locally modified.